lib/rust/flexer/tests/flexer_generated_api_test.rs

```diff
@@ -0,0 +1,517 @@
+                self.gen_group_1_rule_0();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                FlexerStageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_state_1_to_7(&mut self) -> FlexerStageStatus {
+        match u32::from(self.reader.character()) {
+            98 => FlexerStageStatus::ContinueWith(7),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_1();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                FlexerStageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_group_1_rule_0(&mut self) {
+        self.def_on_spaced_word_str();
+    }
+
+    fn gen_group_1_rule_1(&mut self) -> () {
+        self.def_on_spaced_word_str();
+    }
+
+    fn gen_group_1_rule_2(&mut self) {
+        self.def_on_no_err_suffix();
+    }
+
+    fn gen_group_1_rule_3(&mut self) {
+        self.def_on_err_suffix()
+    }
+}
+
+
+// === Trait Impls ===
+
+impl<Reader:LazyReader> Deref for TestLexer<Reader> {
+    type Target = Flexer<TestState,AST,Reader>;
+    fn deref(&self) -> &Self::Target {
+        &self.lexer
+    }
+}
+
+impl<Reader:LazyReader> DerefMut for TestLexer<Reader> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.lexer
+    }
+}
+
+
+
+// ===================
+// === Lexer State ===
+// ===================
+
+/// The stateful components of the test lexer.
+#[derive(Debug)]
+pub struct TestState {
+    /// The initial state of the lexer.
+    initial_state: Rc<Group>,
+    /// The state entered when the first word has been seen.
+    seen_first_word_state: Rc<Group>,
+    /// A bookmark that is set when a match occurs, allowing for rewinding if necessary.
+    matched_bookmark: BookmarkId,
+}
+
+
+// === Trait Impls ===
+
+impl <Reader:LazyReader> FlexerState<Reader> for TestState {
+    fn new(reader:&mut Reader) -> Self {
+        let initial_state         = Rc::new(Group::new(0,String::from("ROOT"),None));
+        let seen_first_word_state = Rc::new(Group::new(1,String::from("SEEN FIRST WORD"),None));
+        let matched_bookmark      = reader.add_bookmark();
+        Self{initial_state,seen_first_word_state,matched_bookmark}
+    }
+
+    fn initial_state(&self) -> &Group {
+        &self.initial_state
+    }
+
+    fn groups(&self) -> Vec<&Group> {
+        vec![&self.initial_state,&self.seen_first_word_state]
+    }
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+/// Executes the test on the provided input string slice.
+fn run_test_on(str:&str) -> Vec<AST> {
+    // Hardcoded for ease of use here.
+    let reader = Reader::new(str.as_bytes(),DecoderUTF8());
+    let lexer: Flexer<TestState,AST,Reader<DecoderUTF8,&[u8]>> = Flexer::new(reader);
+    let mut lexer = TestLexer {lexer};
+
+    match lexer.run() {
+        FlexerResult::Success(tokens) => tokens,
+        _ => vec![]
```

Comment:
use defaults in such places. The code is better when refacotirng then. If you change vec to map or something else, you will need to refactor smaller num of lines